File: python-examples/01-agent-with-tools/agents.py
"""Example workflows and agents for the Hello World example."""

from polos import Agent, max_steps, MaxStepsConfig
from tools import get_weather


# Define a weather agent that can look up weather information
weather_agent = Agent(
    id="weather_agent",
    provider="openai",
    model="gpt-4o-mini",
    system_prompt="You are a helpful weather assistant. Use the get_weather tool to look up weather information when asked.",
    tools=[get_weather],
    stop_conditions=[
        max_steps(MaxStepsConfig(limit=10)),
    ],
)

File: typescript-examples/01-agent-with-tools/agents.ts
/**
 * Example workflows and agents for the Hello World example.
 */

import { defineAgent, maxSteps } from '@polos/sdk';
import { openai } from '@ai-sdk/openai';
import { getWeather } from './tools.js';

// Define a weather agent that can look up weather information
export const weatherAgent = defineAgent({
  id: 'weather_agent',
  model: openai('gpt-4o-mini'),
  systemPrompt:
    'You are a helpful weather assistant. Use the get_weather tool to look up weather information when asked.',
  tools: [getWeather],
  stopConditions: [maxSteps({ count: 10 })],
});
